docs/concepts.md

```diff
@@ -0,0 +1,123 @@
+# Concepts
+
+PrairieLearn is extremely flexible and powerful, but this flexibility can be overwhelming at first. This document provides an overview of the key concepts in PrairieLearn, and how they fit together.
```

Comment:
I'm torn on the first sentence here. Acknowledging that folks might feel overwhelmed is empathetic, but I don't know that we benefit from describing our product as "overwhelming". We can probably drop the first sentence without losing anything, but let's discuss this.

Reply:
can appear complex?

Reply:
I have some tweaks in the latest version.